plugins/hls-hlint-plugin/hls-hlint-plugin.cabal

```diff
@@ -24,6 +24,12 @@ flag ghc-lib
   description:
     Force dependency on ghc-lib-parser even if GHC API in the ghc package is supported
 
+flag hlint33
+  default:     True
+  manual:      False
+  description:
+    Hlint-3.3 doesn't support older ghc/ghc-lib versions, so we can use hlint-3.2 for backwards compat
+
```

Comment:
I presume this is required in order to support hlint on ghc. 
@jneira what are the use cases for that, can we drop it?

Reply:
Yeah hlint-3.3+ghc-lib-9.0 should work for all ghc versions (like hlint<3.2+ghc-lib-8.1.0 worked for 8.8 and 8.6)

Reply:
The reason I added this flag is that none of the formatters support ghc-lib-9.0 yet, so we would get conflicts if we would try using hlint-3.3 together with any of the formatting plugins. As soon as they support ghc9, this flag can be removed.

Reply:
thanks, could we add the note about why it is needed and when we could remove it in the description?

Reply:
Out of curiosity, have you tried to use `hlint ^> 3.2 || hlint ^> 3.3` to see if cabal is able to choose the right version based in other constraints?

Reply:
There's also the problem that hlint has dropped support for ghc-8.6:
https://github.com/ndmitchell/hlint/commit/4df9a6fdaca867d82e46e57d5ddce1a577447d86#diff-3ab702ffba9192c29e2d49f82e65c36af604ea811daed8644bf3037fb6f4fc11

Reply:
If we could find a good way to set the flag:
```
#ifdef HLINT_ON_GHC_LIB
```
we could probably remove most/all of this complexity.

Reply:
Is the new description here clear enough?

Reply:
Looks good to me